src/gui/properties/propertieswidget.cpp

```diff
@@ -311,8 +311,11 @@ void PropertiesWidget::loadTorrentInfos(BitTorrent::TorrentHandle *const torrent
       label_created_by_val->setText(m_torrent->creator());
 
       // List files in torrent
-      PropListModel->model()->setupModelData(m_torrent->info());
-      filesList->setExpanded(PropListModel->index(0, 0), true);
+      BitTorrent::TorrentInfo info = m_torrent->info();
+      libtorrent::file_storage files = info.files();
+      PropListModel->model()->setupModelData(info);
+      if (!(info.filesCount() > 1 && files.name().empty()))
```

Comment:
>I assume you want a new wrapped type? Wouldn't that need to also extend TorrentInfo to return that type?

We shouldn't create wrapper for libtorrent::file_storage since we don't (really) want it used outside abstraction layer (we do not allow arbitrary changes of files mapping).
>Or extend TorrentInfo with a new function that will do the check internally? eg

It's already better. But, actually, it does not fully reflect the essence of this feature.
As we decide to create a subfolder or not, when you add a torrent and we are not changing it anywhere else, hence this concept should not exist anywhere else. That is, we must accept this torrent so if this subfolder it does not and never has been, instead of trying to hide something (I mean subfolder with empty name).
Probably I'm speaking of here is far from real implementations, so it may differ in details. I'm just trying to convey the General meaning.